benchmarks/ddp/README.md

```diff
@@ -0,0 +1,112 @@
+
+It optionally produces a JSON file with all measurements, allowing for
+an easy A/B comparison of code, configuration, or environment. This
+comparison can be produced by `diff.py`.
+
+## Requirements
+
+This benchmark depends on PyTorch and torchvision.
+
+## How to run
+
+Run as many copies of this script as you have model replicas.
+
+If you launch a single task per machine with multiple GPUs, consider
+using [`torch.distributed.launch`][launch] to spawn multiple processes
+per machine.
+
+[launch]: https://pytorch.org/docs/stable/distributed.html#launch-utility
+
+Example output (only on rank 0):
+
+```
+-----------------------------------
+PyTorch distributed benchmark suite
+-----------------------------------
+
+* PyTorch version: 1.4.0a0+fcb7371
+* CUDA version: 10.0
+
+--- nvidia-smi topo -m ---
+
+        GPU0    GPU1    GPU2    GPU3    GPU4    GPU5    GPU6    GPU7    mlx5_2  mlx5_0  mlx5_3  mlx5_1  CPU Affinity
+GPU0     X      NV1     NV1     NV2     NV2     SYS     SYS     SYS     SYS     PIX     SYS     PHB     0-19,40-59
+GPU1    NV1      X      NV2     NV1     SYS     NV2     SYS     SYS     SYS     PIX     SYS     PHB     0-19,40-59
+GPU2    NV1     NV2      X      NV2     SYS     SYS     NV1     SYS     SYS     PHB     SYS     PIX     0-19,40-59
+GPU3    NV2     NV1     NV2      X      SYS     SYS     SYS     NV1     SYS     PHB     SYS     PIX     0-19,40-59
+GPU4    NV2     SYS     SYS     SYS      X      NV1     NV1     NV2     PIX     SYS     PHB     SYS     0-19,40-59
+GPU5    SYS     NV2     SYS     SYS     NV1      X      NV2     NV1     PIX     SYS     PHB     SYS     0-19,40-59
+GPU6    SYS     SYS     NV1     SYS     NV1     NV2      X      NV2     PHB     SYS     PIX     SYS     0-19,40-59
+GPU7    SYS     SYS     SYS     NV1     NV2     NV1     NV2      X      PHB     SYS     PIX     SYS     0-19,40-59
+mlx5_2  SYS     SYS     SYS     SYS     PIX     PIX     PHB     PHB      X      SYS     PHB     SYS
+mlx5_0  PIX     PIX     PHB     PHB     SYS     SYS     SYS     SYS     SYS      X      SYS     PHB
+mlx5_3  SYS     SYS     SYS     SYS     PHB     PHB     PIX     PIX     PHB     SYS      X      SYS
+mlx5_1  PHB     PHB     PIX     PIX     SYS     SYS     SYS     SYS     SYS     PHB     SYS      X
+
+Legend:
+
+  X    = Self
+  SYS  = Connection traversing PCIe as well as the SMP interconnect between NUMA nodes (e.g., QPI/UPI)
+  NODE = Connection traversing PCIe as well as the interconnect between PCIe Host Bridges within a NUMA node
+  PHB  = Connection traversing PCIe as well as a PCIe Host Bridge (typically the CPU)
+  PXB  = Connection traversing multiple PCIe switches (without traversing the PCIe Host Bridge)
+  PIX  = Connection traversing a single PCIe switch
+  NV#  = Connection traversing a bonded set of # NVLinks
+
+--------------------------
+
+
+Benchmark: resnet50 with batch size 32
+   1 GPUs --   warmup: p50:  0.098   327/s  p75:  0.098   326/s  p90:  0.098   326/s  p95:  0.098   326/s
```

Comment:
it would be appreciated to explain the metric is for per GPU metric, and what does "0.098" mean and what does "327/s" mean?

Reply:
Yes, column headers would be nice for this output. Also, would be nice to mention the unit for latency (ex: 0.098)

Reply:
Updated:

```
                            sec/iter    ex/sec      sec/iter    ex/sec      sec/iter    ex/sec      sec/iter    ex/sec
   1 GPUs --   warmup:  p50:  0.097s     328/s  p75:  0.097s     328/s  p90:  0.098s     327/s  p95:  0.098s     327/s
   1 GPUs --    1M/1G:  p50:  0.097s     329/s  p75:  0.097s     329/s  p90:  0.097s     329/s  p95:  0.097s     328/s
   2 GPUs --    1M/2G:  p50:  0.103s     310/s  p75:  0.103s     309/s  p90:  0.103s     309/s  p95:  0.104s     307/s
   4 GPUs --    1M/4G:  p50:  0.103s     310/s  p75:  0.103s     310/s  p90:  0.103s     310/s  p95:  0.103s     310/s
   8 GPUs --    1M/8G:  p50:  0.104s     307/s  p75:  0.104s     306/s  p90:  0.104s     306/s  p95:  0.104s     306/s
  16 GPUs --    2M/8G:  p50:  0.104s     306/s  p75:  0.105s     306/s  p90:  0.105s     305/s  p95:  0.106s     302/s
```